src/arrayMethodSort.js

```diff
@@ -4,8 +4,27 @@
  * Implement method Sort
  */
 function applyCustomSort() {
-  [].__proto__.sort2 = function(compareFunction) {
-    // write code here
+  [].__proto__.sort2 = function (
+    compareFunction = (a, b) => {
+      if (a + '' < b + '') {
+        return -1;
+      }
+
+      return 1;
+    },
+  ) {
+    for (let i = 0; i < this.length; i++) {
+      for (let j = 0; j < this.length; j++) {
+        if (i !== j && compareFunction(this[i], this[j]) < 0) {
+          const temp = this[j];
+
+          this[j] = this[i];
+          this[i] = temp;
+        }
+      }
+    }
+
+    return this;
   };
```

Comment:
Extending the Array prototype is generally discouraged as it can lead to unexpected behavior when other scripts assume the prototype has not been modified. Consider defining a standalone function instead.